rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -344,7 +345,8 @@ impl AccumulationQueue {
             // Push into buffered_arrays without copy since we are about to flush anyways
             self.buffered_arrays.push(array);
             self.current_bytes = 0;
-            Some(std::mem::take(&mut self.buffered_arrays))
+            Some(self.slice_all_buffered_arrays_into_flush_task_arrays())
```

Comment:
when doing `AccumulationQueue::insert`, I think it's better to have the last portion of the `arrow array` remain in `self.buffered_arrays`?